pkg/platform/platform.go

```diff
@@ -234,3 +235,21 @@ func isMachineConfigClusterOperatorReady(client cnoclient.Client) (bool, error)
 	machineConfigClusterOperatorReady := available && !degraded && !progressing
 	return machineConfigClusterOperatorReady, nil
 }
+
+func GetMachineConfigPoolStatuses(ctx context.Context, client cnoclient.Client, mcLabel string) ([]mcfgv1.MachineConfigPoolStatus, error) {
+	mcpList := &mcfgv1.MachineConfigPoolList{}
+	if err := client.Default().CRClient().List(ctx, mcpList); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+	}
+	var mcpStatuses []mcfgv1.MachineConfigPoolStatus
+	for _, mcp := range mcpList.Items {
+		for lKey, lValue := range mcp.Spec.MachineConfigSelector.MatchLabels {
```

Comment:
What if `MatchExpressions` is used instead of `MatchLabels`?
```
  machineConfigSelector:
    matchExpressions:
      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,infra]}
```
from  https://github.com/openshift/machine-config-operator/blob/master/docs/custom-pools.md

I guess we need to use some kind of kubernetes library for label matching here.